Transit plaintext backup breaks audit logging for read key operations

<!-- Please reserve GitHub issues for bug reports and feature requests.

For questions, the best place to get answers is on our [mailing list](https://groups.google.com/forum/#!forum/vault-tool), as they will get more visibility from experienced users than the issue tracker.

Please note: We take Vault's security and our users' trust very seriously. If you believe you have found a security issue in Vault, please responsibly disclose by contacting us at security@hashicorp.com. Our PGP key is available at [our security page](https://www.hashicorp.com/security/).
-->


<!-- Uncomment this section if this is a feature request. Include or exclude other sections as deemed appropriate.
**Feature Request:**
-->

**Environment:**
<!-- The version can be retrieved with `vault version`. -->

* Vault Version: reproducable with 0.9.1 and 0.9.3
* Operating System/Architecture: linux amd64

**Vault Config File:**
```
storage "consul" {
   address="127.0.0.1:8500"
   path="vault"
}

listener "tcp" {
  address    = "127.0.0.1:8200"
  tls_key_file = "/etc/vault/local_server.key"
  tls_cert_file = "/etc/vault/local_server.crt"
}

disable_mlock = true
max_lease_ttl = "900h"
```


**Expected Behavior:**
Creating a backup should not break read key operations.

**Actual Behavior:**
After creating a key backup, reading the key returns error 500:
```
$ vault read transit/dummy/keys/verysecret  
Error reading transit/dummy/keys/verysecret: Error making API request.

URL: GET https://127.0.0.1:8200/v1/transit/dummy/keys/verysecret
Code: 500. Errors:

* internal error
```

Log output:
```
2018/02/06 07:26:14.495724 [ERROR] audit: backend failed to log response: backend=file/ error=time.Time value in a non map key cannot be hashed for audits
2018/02/06 07:26:14.495913 [ERROR] core: failed to audit response: request_path=transit/dummy/keys/verysecret error=1 error occurred:

* no audit backend succeeded in logging the response
```


**Steps to Reproduce:**
```
vault audit enable file file_path=/tmp/awesome_audit_log.log
vault mount -path=transit/dummy transit
vault write transit/dummy/keys/verysecret exportable=true allow_plaintext_backup=true
vault read transit/dummy/backup/verysecret
vault read transit/dummy/keys/verysecret
```

**Important Factoids:**
The error is persistent across application restarts, which means creating a backup for a key seems to persist faulty information in the key metadata (i would suspect the BackupInfo/RestoreInfo structs cause this issue).

--- a/builtin/logical/transit/path_keys.go
+++ b/builtin/logical/transit/path_keys.go
@@ -211,12 +211,16 @@ func (b *backend) pathPolicyRead(ctx context.Context, req *logical.Request, d *f
 	}
 
 	if p.BackupInfo != nil {
-		resp.Data["backup_time"] = p.BackupInfo.Time
-		resp.Data["backup_version"] = p.BackupInfo.Version
+		resp.Data["backup_info"] = map[string]interface{}{
+			"time":    p.BackupInfo.Time,
+			"version": p.BackupInfo.Version,
+		}
 	}
 	if p.RestoreInfo != nil {
-		resp.Data["restore_time"] = p.RestoreInfo.Time
-		resp.Data["restore_version"] = p.RestoreInfo.Version
+		resp.Data["restore_info"] = map[string]interface{}{
+			"time":    p.RestoreInfo.Time,
+			"version": p.RestoreInfo.Version,
+		}
 	}
 
 	if p.Derived {

--- a/builtin/logical/transit/path_keys.go
+++ b/builtin/logical/transit/path_keys.go
@@ -207,9 +207,16 @@ func (b *backend) pathPolicyRead(ctx context.Context, req *logical.Request, d *f
 			"supports_decryption":    p.Type.DecryptionSupported(),
 			"supports_signing":       p.Type.SigningSupported(),
 			"supports_derivation":    p.Type.DerivationSupported(),
-			"backup_info":            p.BackupInfo,
-			"restore_info":           p.RestoreInfo,
 		},
+	}
+
+	if p.BackupInfo != nil {
+		resp.Data["backup_time"] = p.BackupInfo.Time
+		resp.Data["backup_version"] = p.BackupInfo.Version
+	}
+	if p.RestoreInfo != nil {
+		resp.Data["restore_time"] = p.RestoreInfo.Time
+		resp.Data["restore_version"] = p.RestoreInfo.Version
 	}
 
 	if p.Derived {

